src/main/java/org/pac4j/vertx/core/engine/VertxSecurityLogic.java

```diff
@@ -13,16 +13,21 @@
    See the License for the specific language governing permissions and
    limitations under the License.
  */
-package org.pac4j.vertx;
+package org.pac4j.vertx.core.engine;
+
+import org.pac4j.core.engine.DefaultSecurityLogic;
+import org.pac4j.core.profile.ProfileManager;
+import org.pac4j.vertx.VertxProfileManager;
+import org.pac4j.vertx.VertxWebContext;
 
 /**
  * @author Jeremy Prime
  * @since 2.0.0
  */
-public interface TestConstants {
-    // User 1, with no permissions
-    public static final String USER_1 = "testUser1";
-    // User 2, with the permission permission1
-    public static final String USER_2 = "testUser2";
-    public static final String USER_2_PERMISSION = "permission1";
+public class VertxSecurityLogic extends DefaultSecurityLogic<Void, VertxWebContext> {
```

Comment:
I'll add an issue for removing the customized logics so it doesn't get forgotten but will make it dependent on 1.9.2 or later. At least we will have a path for removing the customized logic classes. I'm ok with a customized ProfileManager because that's kind of a behaviour we expect may have to vary per implementation, the same with the actual handlers and the WebContext. Some underlying stuff around CAS logout will also be per-implementation, but I think you have that quite well-isolated. If we can isolate some more, so much the better (it rather feels like there should be a vertx-pac4j-cas library so we don't have to have cas implementations where people aren't using cas, but I'm a bit concerned about library proliferation!)